app/controllers/webhooks/stripe/base_controller.rb

```diff
@@ -0,0 +1,35 @@
+class Webhooks::Stripe::BaseController < ApplicationController
+  include StripeWebhookEvents
+
+  skip_before_action :verify_authenticity_token
+  protect_from_forgery with: :null_session
+
+  before_action :authenticate
+  before_action :validate_event_type
+  before_action :validate_event_environment if Rails.env.production?
+
+  rescue_from Stripe::APIConnectionError, with: :stripe_error
+  rescue_from Stripe::StripeError, with: :stripe_error
+
+  def stripe_error
+    head status: 200
+  end
+
+  protected  
+
+  def validate_event_type
+    if !StripeWebhookEvents::VALUES.include?(params[:type])
+      stripe_error
```

Comment:
We should just render

```ruby
head :ok
```

instead of raising an error. This isn't really an error; it's just something we don't want to process.